controllers/syncer/csi_node.go

```diff
@@ -81,8 +81,12 @@ func (s *csiNodeSyncer) SyncFn() error {
 
 func (s *csiNodeSyncer) ensurePodSpec() corev1.PodSpec {
 	return corev1.PodSpec{
-		Containers:         s.ensureContainersSpec(),
-		Volumes:            s.ensureVolumes(),
+		Containers: s.ensureContainersSpec(),
+		Volumes:    s.ensureVolumes(),
+		SecurityContext: &corev1.PodSecurityContext{
+			RunAsNonRoot: func(b bool) *bool { return &b }(true),
+			RunAsUser:    func(uid int64) *int64 { return &uid }(2121),
+		},
 		ServiceAccountName: constants.GetResourceName(constants.CSINodeServiceAccount),
```

Comment:
Since node pods are running as non root in Docker file, we have to mention a non root user in Daemonser spec as well